File: Popcorn/Domain/Models/MovieGenre.swift
//
//  MovieGenre.swift
//  Popcorn
//
//  Created by Eduardo Raupp Peretto on 09/05/25.
//

import Foundation

struct MovieGenre: SelectionPickable {
    let id: Int
    let name: String
}

extension MovieGenre: CaseIterable {
    static var allCases: [MovieGenre] {
        return [
            MovieGenre(id: 28, name: "Action"),
            MovieGenre(id: 12, name: "Adventure"),
            MovieGenre(id: 16, name: "Animation"),
            MovieGenre(id: 35, name: "Comedy"),
            MovieGenre(id: 80, name: "Crime"),
            MovieGenre(id: 99, name: "Documentary"),
            MovieGenre(id: 18, name: "Drama"),
            MovieGenre(id: 10751, name: "Family"),
            MovieGenre(id: 14, name: "Fantasy"),
            MovieGenre(id: 36, name: "History"),
            MovieGenre(id: 27, name: "Horror"),
            MovieGenre(id: 10402, name: "Music"),
            MovieGenre(id: 9648, name: "Mystery"),
            MovieGenre(id: 10749, name: "Romance"),
            MovieGenre(id: 878, name: "Science Fiction"),
            MovieGenre(id: 10770, name: "TV Movie"),
            MovieGenre(id: 53, name: "Thriller"),
            MovieGenre(id: 10752, name: "War"),
            MovieGenre(id: 37, name: "Western")
        ]
    }
}

// MARK: - MovieGenre + Equatable

extension MovieGenre: Equatable {}

File: KotlinShared/shared/src/commonMain/kotlin/com/eduardoperetto/popcornshared/domain/MovieGenre.kt
package com.eduardoperetto.popcornshared.domain

data class MovieGenre(
    override val name: String,
    val id: Int
) : SelectionPickable {
    companion object {
        val allCases: List<MovieGenre> = listOf(
            MovieGenre("Action", 28),
            MovieGenre("Adventure", 12),
            MovieGenre("Animation", 16),
            MovieGenre("Comedy", 35),
            MovieGenre("Crime", 80),
            MovieGenre("Documentary", 99),
            MovieGenre("Drama", 18),
            MovieGenre("Family", 10751),
            MovieGenre("Fantasy", 14),
            MovieGenre("History", 36),
            MovieGenre("Horror", 27),
            MovieGenre("Music", 10402),
            MovieGenre("Mystery", 9648),
            MovieGenre("Romance", 10749),
            MovieGenre("Science Fiction", 878),
            MovieGenre("TV Movie", 10770),
            MovieGenre("Thriller", 53),
            MovieGenre("War", 10752),
            MovieGenre("Western", 37)
        )
    }
}